js/style/style.js

```diff
@@ -446,6 +446,11 @@ Style.prototype = util.inherit(Evented, {
     },
 
     _forwardTileEvent: function(e) {
+        if (e.type === 'tile.load') {
+            if (this.loaded()) {
+                this.fire('tiles.loaded');
+            }
+        }
```

Comment:
This is a good strategy for triggering the event. `this.loaded()` also deals with sprites, therefore:

 - We should do a similar check on the sprite `load` event. There are pathological cases were we might miss firing `tiles.loaded` because some icons or glyphs were pending.
 - `tiles.load` isn't the best name because this event also indicates something about sprite state (and maybe should be extended to indicate the state of image and video assets in a later PR). See https://github.com/mapbox/mapbox-gl/issues/5 for a discussion in naming. (I'm strongly against any word that's a direct synonym of "load" but I defer to your judgement call.)